projects/packages/videopress/src/client/block-editor/blocks/videopress/components/videopress-uploader/index.js

```diff
@@ -227,16 +252,60 @@ const VideoPressUploader = ( { attributes, setAttributes, noticeUI, noticeOperat
 			if ( getGuidFromVideoUrl( videoUrl ) ) {
 				return onSelectURL( videoUrl );
 			}
-		}
+		} else if ( media.id ) {
+			setIsLoading( true );
+			const path = `videopress/v1/upload/${ media.id }`;
+			apiFetch( { path, method: 'GET' } )
+				.then( result => {
+					setIsLoading( false );
+					if ( 'new' === result.status || 'resume' === result.status ) {
+						setFile( {
+							size: result.file_size,
+							name: result.file_name,
+						} );
+						setUploadingProgress( result.bytes_uploaded, result.file_size );
+						startUploadFromLibrary( media.id );
+					} else if ( 'uploaded' === result.status ) {
+						const videoUrl = `https://videopress.com/v/${ result.uploaded_video_guid }`;
+						return onSelectURL( videoUrl );
+					} else {
+						setUploadErrorDataState( {
+							data: {
+								message: __( 'Error selecting video. Please try again.', 'jetpack-videopress-pkg' ),
+							},
+						} );
+						return;
+					}
+				} )
+				.catch( error => {
+					setUploadErrorDataState( {
+						data: { message: error.message },
+					} );
+					setIsLoading( false );
+					return;
+				} );
+		} else if ( fileUrl ) {
+			if ( ! isBlobURL( fileUrl ) ) {
+				setUploadErrorDataState( {
+					data: { message: __( 'Please select a VideoPress video', 'jetpack-videopress-pkg' ) },
+				} );
+				return;
+			}
 
-		setUploadErrorDataState( {
-			data: {
-				message: __(
-					'Please select a VideoPress video from Library or upload a new one',
-					'jetpack-videopress-pkg'
-				),
-			},
-		} );
+			const file = getBlobByURL( fileUrl );
+			const isResumableUploading = null !== file && file instanceof File;
```

Comment:
`file instanceof File` should be enough, as `null instanceOf File` is `false`.

Reply:
ok. This actually comes from the old code in the old block. I can change it

Reply:
https://github.com/Automattic/jetpack/pull/25792/commits/d7423343003b94eb4ace79c705da53a05e3c9e64